libpromises/evalfunction.c

```diff
@@ -5442,9 +5443,10 @@ static FnCallResult FnCallFormat(EvalContext *ctx, ARG_UNUSED const Policy *poli
                     }
                     else if (strrchr(format_piece, 's'))
                     {
-                        snprintf(piece, CF_BUFSIZE, format_piece, data);
-                        BufferAppend(buf, piece, strlen(piece));
-                        // CfOut(OUTPUT_LEVEL_INFORM, "", "format: appending string format piece = '%s' with data '%s'", format_piece, data);
+                        char *formated;
+                        xasprintf(&formated, format_piece, data);
+                        BufferAppend(buf, formated, strlen(formated));
+                        free(formated);
```

Comment:
There is `BufferAppendF()` which does the same as the functions above.

Reply:
Nice @vpodzime !